duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -100,22 +106,28 @@ public Response executeHttpRequest() throws Exception {
             + method);
       }
 
-      Request request = builder.url(url)
-          .build();
+      // finish and execute request
+      Request request = builder.headers(headers.build()).url(url).build();
+      return executeRequest(request);
+    }
 
-      // Set up client.
-      OkHttpClient httpclient = new OkHttpClient();
-      if (proxy != null) {
-        httpclient.setProxy(proxy);
+    private Response executeRequest(Request request) throws Exception {
+      Response response = httpClient.newCall(request).execute();
+      int attempts = 1;
+      while (attempts < MAX_REQUEST_ATTEMPTS && response.code() == 429) {
+        sleep(Math.min(
+          (int) (Math.pow(BACKOFF_FACTOR, (attempts - 1)) * 1000) + random.nextInt(1000),
```

Comment:
I think we want the random offset outside of the Math.min(), so that way we are always adding some random amount of time.

Reply:
That sounds reasonable to me, though then "MAX_BACKOFF_MS" won't _actually_ be the maximum possible backoff, though that's just a nitpick.